src/parse-result.js

```diff
@@ -63,6 +66,14 @@ export function namespace(options) {
     }
   }
 
+  [BooleanElement, StringElement, NumberElement, ObjectElement, ArrayElement].forEach((item) => {
+    Object.defineProperty(item.prototype, 'sourceMap', {
+      get() {
+        return this.attributes.get('sourceMap');
+      },
```

Comment:
No. I don't think we should add a setter here. You can set it by actually doing `attributes.sourceMap....`, this is just a shortcut.